ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.container.js

```diff
@@ -162,28 +176,20 @@ const mapStateToProps = (state, ownProps) => {
     to: txParamsToAddress,
     gasPrice,
     gas: gasLimit,
-    value: amount,
     data,
   } = (transaction && transaction.txParams) || txParams;
   const accounts = getMetaMaskAccounts(state);
   const smartTransactionsOptInStatus = getSmartTransactionsOptInStatus(state);
   const currentChainSupportsSmartTransactions =
     getCurrentChainSupportsSmartTransactions(state);
 
-  const transactionData = parseStandardTokenTransactionData(data);
-  const tokenToAddress = getTokenAddressParam(transactionData);
-
   const { balance } = accounts[fromAddress];
   const fromInternalAccount = getInternalAccountByAddress(state, fromAddress);
   const fromName = fromInternalAccount?.metadata.name;
   const keyring = findKeyringForAddress(state, fromAddress);
 
-  const isSendingAmount =
-    type === TransactionType.simpleSend || !isEmptyHexString(amount);
-
-  const toAddress = isSendingAmount
-    ? txParamsToAddress
-    : propsToAddress || tokenToAddress || txParamsToAddress;
+  const tokenToAddress = getTokenToAddress(data, type);
+  const toAddress = propsToAddress || tokenToAddress || txParamsToAddress;
```

Comment:
Looks like getTokenToAddress is a good improvement for transfer methods. I am concerned we may be reverting back another issue where sending tokens along with other method calls (approvals) will display the to address instead of the contract address

ref:
- https://github.com/MetaMask/metamask-extension/issues/18800
- https://github.com/MetaMask/metamask-extension/pull/18652#issuecomment-1519635122
- https://github.com/MetaMask/MetaMask-planning/issues/380

Reply:
Hey @digiwand thank you for bringing up this concern.
I've tested some scenarios to check if we are displaying the `to` address instead of the contract address, but the contract address was displayed correctly. Please let me know if there are any particular cases I might not have covered.

[Contract Interaction contract address as the recipient 18800.webm](https://github.com/MetaMask/metamask-extension/assets/45455812/530f52f9-433b-42e7-9142-fca095700638)

[Contract Interaction contract address as the recipient 18800-2.webm](https://github.com/MetaMask/metamask-extension/assets/45455812/e7e05257-ffef-4d97-a0cd-41dd3d7a37be)

Reply:
👍🏼  thanks for confirming!